src/frameworks/compose/discover/frameworks.ts

```diff
@@ -0,0 +1,117 @@
+import { Client } from "../../../apiv2";
+import { frameworksOrigin } from "../../../api";
+
+export const API_VERSION = "v1";
+
+const client = new Client({
+  urlPrefix: frameworksOrigin,
+  auth: true,
+  apiVersion: API_VERSION,
+});
+
+export type State = "BUILDING" | "BUILD" | "DEPLOYING" | "READY" | "FAILED";
+
+interface Codebase {
+  repository?: string;
+  rootDirectory: string;
+}
+
+/** A Stack, the primary resource of Frameworks. */
+interface Stack {
+  name: string;
+  mode?: string;
+  codebase: Codebase;
+  labels: Record<string, string>;
+  createTime: string;
+  updateTime: string;
+  uri: string;
+}
+
+export type StackOutputOnlyFields = "createTime" | "updateTime" | "uri";
+
+interface Build {
+  name: string;
+  state: State;
+  error: Status;
+  image: string;
+  source: BuildSource;
+  buildLogsUri: string;
+  createTime: Date;
+  updateTime: Date;
+  sourceRef: string;
+}
+
+export type BuildOutputOnlyFields = "createTime" | "updateTime" | "sourceRef";
+
+interface BuildSource {
+  codeBaseSource?: CodebaseSource;
+}
+
+interface Status {
+  code: number;
+  message: string;
+  details: any[];
```

Comment:
Any chance we can have something more specific than `any` here?

Reply:
In the backend its type is declared as `Any` and I wasn't sure what that type looks like as it comes from backend.